checker.c

```diff
@@ -1,21 +1,52 @@
-#include <stdio.h>
-#include <assert.h>
+#include "checker.h"
+
+float readchargeRate(float chargeRate)
+{
+    if(chargeRate > CHARGE_RATE_MAX)
+    {
+        printf("Charge Rate out of range!\n");
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+float readsoc(float soc)
+{
+    if(soc < SOC_MIN || soc > SOC_MAX)
+    {
+        printf("State of Charge out of range!\n");
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+float readtemperature(float temperature)
+{
+    if((temperature < TEMP_MIN) || (temperature > TEMP_MAX))
+    {
+        printf("Temperature out of range!\n");
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
 
 int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+    int retval=0;
+    retval=readtemperature(temperature);
+    retval=readsoc(soc);
+    retval=readchargeRate(chargeRate);
+    if(retval!=0)
+        return TRUE;
+    else
+        return FALSE;
 }
 
 int main() {
-  assert(batteryIsOk(25, 70, 0.7));
-  assert(!batteryIsOk(50, 85, 0));
+   assert(batteryIsOk(BOUNDARY_VAL_TEMP2, BOUNDARY_VAL_SOC2, BOUNDARY_VAL_CHARGE_RATE2));
```

Comment:
Permutation and combinations of all three parameters should be considered. For example one parameter will have 2 boundary values and 3 in between values, i.e. <, ==, > threshold values.
Does the test cover sufficiently? 
test cases could be strengthened further.

Reply:
Thank you for your update